doc/dfu/quark_dfu_uart.rst

```diff
@@ -0,0 +1,140 @@
+Quark DFU over UART Guide
+#########################
+
+.. contents::
+
+Hardware Setup
+**************
+
+The hardware setup consists of two main steps:
+
+ * Connecting the board to a Linux host using UART.
+ * Having the possibility to connect the DM GPIO pin to GND or to press the DM
+   Button.
+
+The DM GPIO pin is the pin used to put the device into DM mode. The device has
+to be reset while the GPIO is connected to GND. On Quark Developer Boards a
+button connects the DM GPIO to GND if pressed.
+
+Software Setup
+**************
+
```

Comment:
We can just link to the section that covers this on the main README.rst file, right? "For environment setup, please refer to (...)."